src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -34,106 +36,238 @@
         shooter.configNominalOutputForward(0, TIMEOUT);
         shooter.configNominalOutputReverse(0, TIMEOUT);
         shooter.configPeakOutputForward(+1, TIMEOUT);
         shooter.configPeakOutputReverse(-1, TIMEOUT);
         shooter.setNeutralMode(NeutralMode.Coast);
-        shooter.configSupplyCurrentLimit(shooterLimit);
+        shooter.configSupplyCurrentLimit(shooterLimit, TIMEOUT);
+        fl.setDirectory("Shooter");
+        fl.createfile("shooterRPMS");
 
-    }
+        try {
+            shooterRpms = Double.parseDouble(fl.readFile());
+        }
+
+        catch (NullPointerException e) {
+            // e.printStackTrace();
+        }
+
+        catch (NumberFormatException e) {
+            // e.printStackTrace();
+        }
 
-    public void runShooter() {
-        // System.out.println(shooter.getSelectedSensorVelocity()); PRINTS FOR VELOCITY
-        // CONTROL
+    }
 
+    public void main() {
         // Control for FlyWheel
+        ShooterMain();
+        feederMain();
+
+        // Control for lowering intake
+        gathererMain();
+    }
+
+    public void gathererMain() {
+        if (FeatureFlags.doCompressor && FeatureFlags.compressorInitialized) {
+            switch (gathererState) {
+                case gathererUp:
+                    if (oi.manualIntakeButton()) {
+                        lowerIntake();
+                        startIntake(intakeSpeed);
+                    }
+
+                    else {
+                        stopIntake();
+                    }
+                    break;
+                case gathererDown:
+                    if (!oi.manualIntakeButton()) {
+                        stopIntake();
+                    }
+
+                    else {
+                        startIntake(intakeSpeed);
+                    }
+                    break;
+                case holding:
+                    if (oi.manualIntakeButton()) {
+                        stopIntake();
+                        raiseIntake();
+                    }
+
+                    else {
+                        stopIntake();
+                    }
+                    break;
+            }
+        }
+    }
+
+    public void ShooterMain() {
+        System.out.println(shooterRpms);
+        updateManualRPMS();
+
         if (oi.runFlyWheelButtonManual()) {
+            if (checkVision()) {
+                vc.shooterstate = shooterState.ALIGN;
+            }
+
             startShooter(shooterRpms);
-        } else {
+        }
+
+        else if (oi.autoShootButton()) {
+            if (checkVision()) {
+                autoShoot();
+            }
+
+            else {
+                if (checkVision()) {
+                    vc.shooterstate = shooterState.ALIGN;
+                }
+                startShooter(shooterRpms);
+                feederMain();
+            }
+        }
+
+        else {
+            if (checkVision()) {
+                vc.shooterstate = shooterState.ALIGN;
```

Comment:
not sure what this does

Reply:
ShooterMain needs to be completely remade